javapoet/src/main/java/com/palantir/javapoet/TypeSpec.java

```diff
@@ -284,16 +295,29 @@ void emit(CodeWriter codeWriter, String enumName, Set<Modifier> implicitModifier
                 // Push an empty type (specifically without nested types) for type-resolution.
                 codeWriter.pushType(new TypeSpec(this));
 
-                codeWriter.emitJavadoc(javadoc);
+                if (kind == Kind.RECORD && recordConstructor != null) {
+                    codeWriter.emitJavadocWithParameters(javadoc, recordConstructor.parameters());
+                } else {
+                    codeWriter.emitJavadoc(javadoc);
+                }
                 codeWriter.emitAnnotations(annotations, false);
                 codeWriter.emitModifiers(modifiers, Util.union(implicitModifiers, kind.asMemberModifiers));
                 if (kind == Kind.ANNOTATION) {
                     codeWriter.emit("$L $L", "@interface", name);
                 } else {
-                    codeWriter.emit("$L $L", kind.toString(), name);
+                    codeWriter.emit("$L $L", kind.name().toLowerCase(Locale.US), name);
```

Comment:
We shouldn't be using `Locale` other than `ROOT`. Ideally we wouldn't be using `toLowerCase` at all.

Let's clean this up by:
- Adding a `keyword` field to `Kind`.
- Removing the `Kind.toString` implementation and replacing it with a `keyword` method.
- Using that here instead.

This eliminates all the `toLowerCase` usage in this repo.

Reply:
Want me to do that in this PR or a flup?

Reply:
Did it here

Reply:
[973f187](https://github.com/palantir/javapoet/pull/17/commits/973f187718bae3bb3d6789e575ee54e13bd97575)